reconcile/dashdotdb_dvo.py

```diff
@@ -23,29 +23,48 @@ def __init__(self, dry_run, thread_pool_size):
         self.dashdotdb_url = secret_content['url']
         self.dashdotdb_user = secret_content['username']
         self.dashdotdb_pass = secret_content['password']
+        self.chunksize = secret_content.get('chunksize') or '20'
         self.logmarker = "DDDB_DVO:"
 
+    def _chunkify(self, data, size):
+        for i in range(0, len(data), int(size)):
+            yield data[i:i+int(size)]
+
     def _post(self, deploymentvalidation):
         if deploymentvalidation is None:
-            return None
+            return
         cluster = deploymentvalidation['cluster']
-        dvdata = deploymentvalidation['data']
-        LOG.info('%s posting validations for %s', self.logmarker, cluster)
-        response = None
-        if not self.dry_run:
-            endpoint = (f'{self.dashdotdb_url}/api/v1/'
-                        f'deploymentvalidation/{cluster}')
-            response = requests.post(url=endpoint, json=dvdata,
-                                     auth=(self.dashdotdb_user,
-                                           self.dashdotdb_pass),
-                                     timeout=(5, 30))
-            try:
-                response.raise_for_status()
-            except requests.exceptions.RequestException as details:
-                LOG.error('%s error posting %s - %s',
-                          self.logmarker, cluster, details)
-
-        LOG.info('%s cluster %s synced', self.logmarker, cluster)
+        # dvd.data.data.result.[{metric,values}]
+        dvresult = deploymentvalidation.get('data').get('data').get('result')
+        if dvresult is None:
+            return
+        LOG.info('%s Processing (%s) metrics for: %s', self.logmarker,
+                 len(dvresult),
+                 cluster)
+        metrics = list(self._chunkify(dvresult, self.chunksize))
```

Comment:
calling list() here just defeats the generator purpose

Reply:
i thought i had totally missed something, but i've verified up and down it's working how i intended. i'm taking a list of dict and converting it to smaller chunks, then iterating on that list.. combing each element with another (static) list.
```
2021-01-20 14:11:08] [INFO] [dashdotdb_dvo.py:_post:75] - DDDB_DVO: DVO data for app-sre-stage-01 synced to dddb
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:43] - DDDB_DVO: Processing (5342) metrics for: app-sre-stage-01
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:51] - DDDB_DVO: Chunked metrics into (268) elements for: app-sre-stage-01
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:61] - DDDB_DVO: len([data][result]):20
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:61] - DDDB_DVO: len([data][result]):20
{snip}
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:61] - DDDB_DVO: len([data][result]):20
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:61] - DDDB_DVO: len([data][result]):2
[2021-01-20 14:11:09] [INFO] [dashdotdb_dvo.py:_post:75] - DDDB_DVO: DVO data for app-sre-stage-01 synced to dddb
```

Reply:
It's working indeed. All I'm saying is that creating a generator and then casting it into a list defeats its purpose.

Reply:
Maybe I'm using the generator in weird way. I'm using the casting to list() as an intentionally cheap way to trigger iteration. I guess that might not be what one normally expects. I could wrap the post inside of the generator.. but I often find that sometimes harder for folks to read/grok.

Reply:
`for metric in self._chunkify(...):` should be enough